### DISCUSSION TITLE
Smart Grid Schalter

### DISCUSSION THREAD

**Discussion by MichaelGer1410** [Trunks1982/Daikin-Rotex-HPSU-CAN#42] [General]
Guten Morgen,

Vorweg, ich möchte über ein FHEM DOIF die SG an und ausschalten sowie ggfs den SG Modus ändern. Um die Daten vom iObroker zu erhalten nutze ich das FHEM Addon.
Leider erhalte ich nur die Text Sensoren für die SG Funktionen.
Z.B. den Modus Heizen, Bereitschaft erhalte ich als "Switch". Das kann ich aus FHEM heraus schalten.
Mache ich etwas falsch? Oder müsstest du da was ändern damit das funktioniert?
VG Michael

**Comment by Trunks1982** [comment]
Ich hab keine Ahnung von fhem wieso nutzt man sowas überhaupt noch?🤣
Sg kann man ja so umstellen alles andere liegt nicht an meiner programmierung.

**Reply by MichaelGer1410** [reply]
Weil ich damit soweit ganz gut klar komme. Bin kein Programmierer wie du ;)

Von FHEM mal abgesehen, im ioBroker sehe ich bei den SG Dingern auch nur die Text Sensoren ob an oder aus und welcher SG Mode gerade eingestellt ist. Umschalten geht dort auch nicht. 
Würde ja gerne einen Screenshot machen, aber bin unterwegs und die Ansicht ist kacke , zeigt nicht alles an am Handy 😁

**Reply by Trunks1982** [reply]
Es ist einmal sg modus und smard grid da zum umschalten

**Reply by MichaelGer1410** [reply]
Ja aber nur über das WEBIF des ESP. Über iObroker werden nur der Status abgefragt. Für schalten fehlen die Schalter.
Auf den ersten beiden bildern ist der Modus für den Modus wie die Heizung gerade läuft. Das ist in dem Baum unter SELECT aufgeführt und man kann den Modus über iobroker ändern.

Die Smart Grid funktionen sind nur unter Text Sensoren aufgeführt und man hat keine Möglichkeit die extern über iobroker mit dem ESPHome Addon zu schlaten, nur über webif.
![Betriebsmodus](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/170225415/52f7edc8-db8f-46c0-a7b6-a4b514182115)
![Betriebsmosus2](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/170225415/1b93efae-5d31-4326-89c3-f84457db3c95)
![smartgrid](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/170225415/ab42a744-0038-4bbe-a16f-8a1f4a57cb5d)
![esprotex](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/170225415/a40dfaf6-cd22-4cb5-880e-d86a62963892)

**Reply by Trunks1982** [reply]
Ich kann selber nachher mal schaun in iob.

**Reply by MichaelGer1410** [reply]
Das wäre lieb :)

**Reply by Trunks1982** [reply]
Siehst du die hk funktion zum verstellen?

**Reply by MichaelGer1410** [reply]
Nein, ist ebenfalls nur bei den "Text Sensoren" aufgeführt.

**Reply by MichaelGer1410** [reply]
"Rotex ESP Restart" ist das um den ESP neuzustarten?

**Reply by Trunks1982** [reply]
Jup.Also kann mir nur vorstellen das es in iob hakt.In HA ist es drin.lösch mal die ganzen datenpunkte und lern in iob neu an.

**Reply by MichaelGer1410** [reply]
Dann änder ich die IP des ESP und füge den neu hinzu.

**Reply by Trunks1982** [reply]
Ja probier das mal

**Reply by MichaelGer1410** [reply]
So, den ganzen datenpunkt (baum) gelöscht und neu angelegt mit anderer IP. Alle wie vorher beschrieben ;)

**Reply by Trunks1982** [reply]
Ok teste es selber mal

**Reply by MichaelGer1410** [reply]
```
text_sensor:
  - platform: template
    name: "Betriebsmodus"
    id: text_betriebsmodus
    lambda: |-
```
```
select:
  - platform: template
    name: "Betriebsmodus setzen"
    id: betrieb
    optimistic: true
```

```
- platform: template
    name: "Smart Grid"
    id: text_smartgrid
    lambda: |-
```

```
 - platform: template
    name: "SG Modus"
    id: sg
```

Kann es sein weil der NAME "Smart Grid" bei beiden einträgen gleich ist und nicht unterschiedlich wie z.B. beim betriebsmodus?
Also "Betriebsmodus" hast du einmal genommen für den textsensor und für select "Betriebsmodus setzen"

das ist bei der SG Funktikon und bei der HK funktion nicht so :)

**Reply by Trunks1982** [reply]
Nein Name ist egal.Id wäre wenn das wichtige.Wie gesagt in HA funzt es so wie es sein soll.Sonst wäre da auch nen Fehler.

**Reply by MichaelGer1410** [reply]
Das ist mir halt aufgefallen was die 4 Modis unterscheidet.
Vielleicht hat iobroker damit nur ein problem?
Könntest du mal eine BIN fertig machen wo es genauso gemacht ist wie bei der Betriebsmodus? Dann teste ich das mal

**Reply by Trunks1982** [reply]
Wie gesagt ich teste das mal

**Reply by MichaelGer1410** [reply]
okay danke!

**Reply by Trunks1982** [reply]
So hab ne neue bin hochgeladen.Sollte jetzt gehen.Der kam sich wohl mit den Namen in die Quere bei IOB.Wie gesagt bei HA hats funktioniert.

**Reply by MichaelGer1410** [reply]
Ah also lag es doch daran? Werde die BIN wenn ichs schaffe später noch flashen. Besten dank!

**Reply by Trunks1982** [reply]
Leider ja, ka warum das bei iob so war bzw ist.

**Reply by MichaelGer1410** [reply]
Funktioniert bei mir jetzt auch. Der vollsändigkeit halber wäre es noch schön wenn die HK Funktion auch noch angepasst wird damit es im iob läuft ;)
